build

```diff
@@ -1,5 +1,18 @@
 #!/usr/bin/env bash
 
+usage() {
+  echo "Usage: $0 [GOOS] [GOARCH]"
+}
+
+if [[ -z "$1" ]] || [[ -z "$2" ]]; then
+    DEFAULT_GOOS=`go env | grep GOOS`
+    DEFAULT_GOARCH=`go env | grep GOARCH`
+    echo "GOOS or GOARCH is omitted, defaulting to $DEFAULT_GOOS $DEFAULT_GOARCH"
+    usage
```

Comment:
I think we should use defaults as proper usage. Meaning lets not echo usage if it is not passed but defaults are populated.

sudo logic
If $1 and $2 -> BUILD_ENV="GOOS=$1 GOARCH=$2" else ... if DEFAULTS don't exist echo usage.

Reply:
per my understanding, there are always DEFAULT GOOS and GOARCH values on a go installed environment. so maybe it will never echo usage.

please correct me for any other changes needed

Reply:
```
$ ./build
GOOS or GOARCH is omitted, defaulting to GOOS="linux" GOARCH="amd64"
Usage: ./build [GOOS] [GOARCH]
```
If I use a script I don't expect to see `usage` unless I did something wrong or uninteded. The echo is fine buts lets drop usage. You could add usage example like how we do in test.

https://github.com/etcd-io/etcd/blob/be39aa5bb2ab93756598339d934c3775eda82507/test#L3-L10

Reply:
Done. have moved usage into annotation